File: frameworks/Java/kora/kora-jdbc-direct/src/main/java/io/koraframework/benchmark/repository/MapperModule.java
package io.koraframework.benchmark.repository;

import ru.tinkoff.kora.common.Module;
import ru.tinkoff.kora.database.jdbc.mapper.parameter.JdbcParameterColumnMapper;

import java.util.Collection;

@Module
public interface MapperModule {

    default JdbcParameterColumnMapper<Collection<Integer>> postgresCollectionOfIntegerJdbcParameterColumnMapper() {
        return (stmt, index, value) -> {
            var typeArray = value.toArray(Integer[]::new);
            var sqlArray = stmt.getConnection().createArrayOf("INT", typeArray);
            stmt.setArray(index, sqlArray);
        };
    }
}

File: frameworks/Kotlin/kora/kora-jdbc-coroutines-direct/src/main/kotlin/io/koraframework/benchmark/repository/MapperModule.kt
package io.koraframework.benchmark.repository

import ru.tinkoff.kora.common.Module
import ru.tinkoff.kora.database.jdbc.mapper.parameter.JdbcParameterColumnMapper

@Module
interface MapperModule {

    fun postgresCollectionOfIntJdbcParameterColumnMapper(): JdbcParameterColumnMapper<Collection<Int>> {
        return JdbcParameterColumnMapper { stmt, index, value ->
            val typeArray = value.toTypedArray()
            val sqlArray = stmt.connection.createArrayOf("INT", typeArray)
            stmt.setArray(index, sqlArray)
        }
    }
}
